### DISCUSSION TITLE
why getting always 32 from cuda::std::countr_zero(unsigned) from cuda-12.3?

### DISCUSSION THREAD

**Discussion by bluebabyxp** [NVIDIA/cccl#2191] [libcu++]
https://godbolt.org/z/Kd63Es7KP

the ptx seems always store 32 to global mem. Seems correct on ctk-12.1.

**Comment by jrhemstad** [comment]
Verified that this looks like a bug in the latest `main` as well: https://github.com/NVIDIA/cccl/issues/2195
